### PR TITLE
Changed code according to new class diagram.

### PR DESCRIPTION
**Pull request by crisvdn**
This feature branch refactores the code according to the new class diagram.

### CODE REVIEW

File: code/headers/beetle.hpp [hunk 1]
```diff
@@ -50,17 +50,5 @@ namespace r2d2::moving_platform {
          * @param    int: new degrees
          */
         void turn(int16_t degrees) override;
-        /**
```
Comment (Joepieler) [inline code/headers/beetle.hpp:53]: Why is there a set_steering and a turn?

File: code/headers/beetle.hpp [hunk 2]
```diff
@@ -50,17 +50,5 @@ namespace r2d2::moving_platform {
...
-         */
-        void move(int8_t distance) override;
-        /**
-         * @brief
-         * This function Will move the platform x forward and than y backwards
-         * @param    int: distance
-         */
-        void move(int8_t x, int8_t y) override;
     };
 } // namespace r2d2::moving_platform
```
Comment (Joepieler) [inline code/headers/beetle.hpp:47]: encoders have the wrong name must be motor_encoder_m0

**Comment by Joepieler** [comment]
Everything is changed according to the diagram.

**Review by Joepieler — APPROVED** [review]
Everything is changed according to the diagram.

**Comment by Lennart99** [comment]
You should also take a look at the constructor of beetle_c
The arguments are changed in code, but not in the documentation
I saw this while scrolling through the code and found this a good place to address this

**Comment by crisvdn** [comment]
Thanks Lennart for pointing this out. Changed it in the last commit.

File: code/headers/beetle.hpp (hunk 2 above)
Comment (crisvdn) [reply code/headers/beetle.hpp:47]: Updated in last commit

File: code/headers/beetle.hpp (hunk 1 above)
Comment (crisvdn) [reply code/headers/beetle.hpp:53]: Set_steering removed.

**Pull request merged**